app/src/main/java/ch/epfl/culturequest/ArtDescriptionDisplayActivity.java

```diff
@@ -259,4 +264,15 @@ private void postImage(String url, BasicArtDescription artwork, List<String> bad
             return null;
         });
     }
+
+    private void shareImage(Uri uri, BasicArtDescription description) {
+        Intent shareIntent = new Intent(Intent.ACTION_SEND);
+        shareIntent.setType("image/jpeg");
+        shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
+        shareIntent.putExtra(Intent.EXTRA_TEXT, "I just scanned " + description.getName() + " with \uD835\uDC02\uD835\uDC2E\uD835\uDC25\uD835\uDC2D\uD835\uDC2E\uD835\uDC2B\uD835\uDC1E\uD835\uDC10\uD835\uDC2E\uD835\uDC1E\uD835\uDC2C\uD835\uDC2D!\n\n" +
+                "It's a " + getRarityLevel(description.getScore()).name().toLowerCase() + " artwork from " + description.getArtist() + ", displayed at "+ description.getMuseum() + ", " + description.getCity() + ".\n\n" +
+                "Download the app here: https://play.google.com/store/apps/details?id=com.culturequest.culturequest");
+        shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        startActivity(Intent.createChooser(shareIntent, "Share your scan using"));
+    }
```

Comment:
So I imagine that this allows you to choose how you want to share it ?

Reply:
The activity intent enables to open the chooser for the method, then you are redirected to the app with the text pasted in your input box, ready to be sent

Reply:
Perfect !